File: src/LeetCode/LeetCode/101.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
  public:
    bool isSymmetric(TreeNode *root) { return isSymmetric(root, root); }

  private:
    bool isSymmetric(TreeNode *a, TreeNode *b) {
        if (not a or not b) {
            return a == b;
        }
        if (a->val == b->val) {
            return isSymmetric(a->left, b->right) and
                   isSymmetric(a->right, b->left);
        }
        return false;
    }
};

File: src/LeetCode/LeetCode/101.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSymmetric(TreeNode root) {
        return IsSymmetric(root, root);
    }

    private bool IsSymmetric(TreeNode a, TreeNode b) {
        if (a == null || b == null) {
            return a == b;
        }
        if (a.val == b.val) {
            return IsSymmetric(a.left, b.right) && IsSymmetric(a.right, b.left);
        }
        return false;
    }
}